storage/remote/queue_manager_test.go

```diff
@@ -413,13 +414,52 @@ func TestReshardPartialBatch(t *testing.T) {
 		case <-done:
 		case <-time.After(2 * time.Second):
 			t.Error("Deadlock between sending and stopping detected")
+			pprof.Lookup("goroutine").WriteTo(os.Stdout, 1)
 			t.FailNow()
 		}
 	}
 	// We can only call stop if there was not a deadlock.
 	m.Stop()
 }
 
+func TestQueueFilledDeadlock(t *testing.T) {
```

Comment:
can we include a short comment of when this deadlock would have been hit without your code change? `a big scrape causing an append to instantly fill a queue at the same time a batch_send_deadline is hit`?

Reply:
:+1: Done.